src/utils/arraysEqual.ts

```diff
@@ -1,17 +1,22 @@
 /**
- * @description 2개의 배열을 비교하는 함수입니다.
+ * @description 2개의 1차원 배열을 순서와 내부 값이 일치하는지 비교하는 함수입니다.
  *
- * @template T 배열 내부 요소의 타입
  *
- * @param T[] array 첫번째 array
- * @param T[] array 두번째 array
+ * @param array1 첫번째 array
+ * @param array2 두번째 array
  *
  * @returns boolean값
  *
- * @example arraysEquial<number>([1,2,3,4,5],[1,2,3,4,1]);
+ * @example isArrayContentEqual([1,2,3,4,5],['a',2,3,1]);
  * //false
  */
-const arraysEqual = <T>(array1: T[], array2: T[]) => {
-  return JSON.stringify(array1) === JSON.stringify(array2);
+const isArrayContentEqual = (array1: any[], array2: any[]): boolean => {
+  if (array1.length !== array2.length) return false;
+  for (let i = 0; i < array1.length; i++) {
+    if (array1[i] !== array2[i]) {
+      return false;
+    }
+  }
+  return true;
```

Comment:
사용하는 곳에서는 1차원 배열의 요소값에 대해서 순서와 값이 일치하는지만 확인하면 될 것 같아서 큰 로직 없이 간단하게 만들어 봤습니다!
함수 이름도  기존 `arrayEqual`에서 내부 값만 비교한다는 뜻인 `isArrayContentEqual`로 변경했습니다.

Reply:
```suggestion
const isEqualArray = <T>(array1: T[] | readonly T[], array2: T[] | readonly T[]): boolean => {
  if (array1.length !== array2.length) return false;
  
  for (let i = 0; i < array1.length; i++) {
    if (array1[i] !== array2[i]) {
      return false;
    }
  }
  return true;
}
```

`isEqualArray` 네이밍은 어떨까요? 좀 더 단순하지만 목적을 충분히 전달합니다.

제네릭 타입(`T`)을 활용하면 좋을 것 같아요 :) `any` 타입보다 더욱 명확합니다.

readonly를 추가 한 이유는 `as const` 배열에도 호환하기 위함입니다.

```ts
// number[] 사용
isEqualArray([1, 2, 3], [3, 4, 5]);

const isEqualArray: <number>(array1: number[] | readonly number[], array2: number[] | readonly number[]) => boolean
```

```ts
// as const 사용
isEqualArray([1, 2, 3] as const, [3, 4, 5] as const);

const isEqualArray: <5 | 1 | 2 | 3 | 4>(array1: (5 | 1 | 2 | 3 | 4)[] | readonly (5 | 1 | 2 | 3 | 4)[], array2: (5 | 1 | 2 | 3 | 4)[] | readonly (5 | 1 | 2 | 3 | 4)[]) => boolean
```

Reply:
내부 요소를 비교한다는 의미에서 `isArrayContentEqual`라는 함수 네임을 붙였는데 @ssi02014 님 말씀대로 `isEqualArray`가 조금 더 심플하면서 비슷한 목적을 전달하는 것 같네요 `내부 요소 비교` 라는 추가 설명은 어차피 주석에 달려있으니 같은 의도가 전달된다면 심플한 쪽을 선택하는게 맞는 것 같습니다!
제네릭 타입 같은 경우에는 초기에 함수를 만들었을 때 사용했습니다.

```ts
const arraysEqual = <T>(array1: T[], array2: T[]) => {
  return JSON.stringify(array1) === JSON.stringify(array2);
};
```
하지만 `@example`쪽 예시처럼 `isEqualArray([1,2,3,4,5],['a','',3,1,true])//false `
와 같이 배열 내부에  다양한 타입이 들어올 수 있다고 생각해서 `any[]`타입으로 변경했습니다
이 부분에 관해서는 @ssi02014님 의견이 궁금합니다부 요소를 비교한다는 의미에서 `isArrayContentEqual`라는 함수 네임을 붙였는데 @ssi02014 님 말씀대로 `isEqualArray`가 조금 더 심플하면서 비슷한 목적을 전달하는 것 같네요 `내부 요소 비교` 라는 추가 설명은 어차피 주석에 달려있으니 같은 의도가 전달된다면 심플한 쪽을 선택하는게 맞는 것 같습니다!
제네릭 타입 같은 경우에는 초기에 함수를 만들었을 때 사용했습니다.

```ts
const arraysEqual = <T>(array1: T[], array2: T[]) => {
  return JSON.stringify(array1) === JSON.stringify(array2);
};
```
하지만 `@example`쪽 예시처럼 `isEqualArray([1,2,3,4,5],['a','',3,1,true])//false `
와 같이 배열 내부에  다양한 타입이 들어올 수 있다고 생각해서 `any[]`타입으로 변경했습니다.
타입이 다른 경우에는 에초에 `false` 값이니 제네릭 T로 array의 타입을 받을지 그럼에도 함수 자체의 목적은 두 배열의 요소가 같은지 아닌지 비교하는것이니 any[]타입으로 둘지  고민입니다.

`array1: number[] | readonly number[], array2: number[] | readonly number[]` 인자를 이와같이 배열이 ` as const`인 경우도 생각해서 유니온 타입으로 받는건 정말 좋은 것 같습니다!

Reply:
@rhehfl 이런 경우에는 함수의 사용케이스를 고민해보면 좋을 것 같아요!

일반적으로 `isEqualArray` 함수를 통해 두 배열의 값을 비교 할 때 유사한 타입의 배열을 비교 할 경우가 많지 완전히 타입이 딴판인 배열을 비교하는 케이스는 적을 것 같은데요!

물론, 예시로 주신 `isEqualArray([1, 2, 3, 4, 5],['a', '', 3, 1, true]); //false` 케이스는 해당 함수가 제대로 동작해요. 왜냐하면 array1/array2의 타입이 `(string | number | boolean)[]`로 타입이 확장되기 때문입니다.

```ts
isEqualArray([1, 2, 3], [1, 2, 4]); // 동일한 타입의 두 배열
isEqualArray([1, 2, 3], [1, 2, '4']); // 2번째 배열에 문자열이 하나 있지만 (string | number)[] 로 확장 될 것이기 때문에 문제 없음
isEqualArray([1, 2, 3], [1, 2, '4', true]);  //  (string | number | boolean)[]
```

Reply:
@rhehfl 타입이 완전히 호환되지 않고, 동떨어진 두 배열의 비교 케이스가 적을 것으로 예상이되고, 특별한 경우가 아니라면 타입 확장으로 잘 동작할 것으로 예상되기 때문에 any보다는 제네릭이 낫지 않을까가 제 의견입니다 ㅎㅎㅎ 👍

Reply:
아!
어차피 제네릭으로 두어도 `(string | number | boolean)[]`처럼 타입 확장이 되니 제네릭으로 두는게 더 좋을 것 같네요
확실히 any보다는 제네릭이 좋을 듯 합니다. 
리뷰 감사합니다!